### DISCUSSION TITLE
Different results from official API

### DISCUSSION THREAD

**Discussion by ghost** [osm-search/Nominatim#2868] [General]
Hi, last week I installed my first instance of Nominatim with a full planet import. But I've seen some inconsistencies between my results and results from https://nominatim.openstreetmap.org/ .

Given this address: `25014 Viking St, Hayward, CA 94545`
And this query: `/search.php?q=25014%20Viking%20St,%20Hayward,%20CA%2094545&format=jsonv2`

The official API returns:

```
[
   {
      "place_id":304739991,
      "licence":"Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
      "osm_type":"way",
      "osm_id":6328881,
      "boundingbox":[
         "37.637818298994",
         "37.637918298994",
         "-122.1248",
         "-122.1247"
      ],
      "lat":"37.63786829899397",
      "lon":"-122.12475",
      "display_name":"25014, Viking Street, Hayward, Comté d'Alameda, Californie, 94545, États-Unis d'Amérique",
      "place_rank":30,
      "category":"place",
      "type":"house",
      "importance":-0.52
   }
]
```

And my instance returns:

```
[
   {
      "place_id":102394391,
      "licence":"Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
      "osm_type":"way",
      "osm_id":6328881,
      "boundingbox":[
         "37.6360982",
         "37.6383799",
         "-122.1249293",
         "-122.1249005"
      ],
      "lat":"37.6373008",
      "lon":"-122.1249191",
      "display_name":"Viking Street, Hayward, Comté d'Alameda, Californie, 94545, États-Unis d'Amérique",
      "place_rank":26,
      "category":"highway",
      "type":"residential",
      "importance":0.6199999999999999
   }
]
```

As you can see even the place rank is different, making the display address less precise (losing house number), what could be the cause of this?

**Comment by mtmail** [comment]
Is the house number in OpenStreetMap? If not you might need to add https://nominatim.org/release-docs/latest/customize/Tiger/

**Reply by ghost** [reply]
Yes OSM does have the house number (the first response query is from OSM, the second is from my Nominatim instance)

**Reply by mtmail** [reply]
I can't see the house number on OSM https://www.openstreetmap.org/way/6328881
![image](https://user-images.githubusercontent.com/3727288/200186327-f5764214-5398-4a76-88b2-227fe9e01edc.png)

**Reply by ghost** [reply]
Oh okay I didn't understand the question correctly, yes its not displayed on OSM but it is displayed on OSM Nominatim. I'll try and install TIGER to see if it fixes it !

**Reply by ghost** [reply]
<img width="1434" alt="Capture d’écran 2022-11-06 à 21 15 13" src="https://user-images.githubusercontent.com/92714362/200192929-a7bc1c07-c255-45fa-b2ca-4cff1975e3a4.png">
I launched the command, it keeps printing a " . " every one second, is this the intended output?

**Reply by mtmail** [reply]
yes, keep waiting

**Reply by ghost** [reply]
<img width="717" alt="Capture d’écran 2022-11-06 à 22 26 00" src="https://user-images.githubusercontent.com/92714362/200196140-69f10903-3df7-4162-b8c8-97a6144536a9.png">
It finished, I restarted the machine but still same results, is there maybe a Nominatim command to refresh something else?

I did this:

`echo NOMINATIM_USE_US_TIGER_DATA=yes >> .env`

`nominatim refresh --functions`

I also have this with `nominatim admin --check-database`

<img width="572" alt="Capture d’écran 2022-11-06 à 22 35 00" src="https://user-images.githubusercontent.com/92714362/200196451-5b365528-1882-4a4d-aa0b-6c66bef1c325.png">

**Reply by ghost** [reply]
I also saw that the documentation references the TIGER for 2021 but 2022 is available ((https://nominatim.org/data/tiger2022-nominatim-preprocessed.csv.tar.gz). I imported it and I still get the same result :/

**Reply by mtmail** [reply]
when you add `&debug=1` to the request, do you see 'tiger' mentioned? https://nominatim.openstreetmap.org/search.php?q=25014%20Viking%20St,%20Hayward,%20CA%2094545&format=jsonv2&debug=1

**Reply by ghost** [reply]
This is the result of the request with `&debug=1` https://hastebin.com/havulazipa.scss

**Reply by ghost** [reply]
It is also much longer than the official one

**Reply by ghost** [reply]
I think the interesting part is this (from my instance):

```
Post-filter results:  0 => 'osm_type' => 'W'
                           'osm_id' => 6328881
                           'class' => 'highway'
                           'type' => 'residential'
                           'admin_level' => 15
                           'rank_search' => 26
                           'rank_address' => 26
                           'place_id' => 102394391
                           'parent_place_id' => 304916188
                           'housenumber' => -1
                           'country_code' => 'us'
                           'langaddress' => 'Viking Street, Hayward, Comté d'Alameda, Californie, 94545, États-Unis d'Amérique'
                           'placename' => 'Viking Street'
                           'ref' => 'Viking Street'
                           'lon' => '-122.1249191'
                           'lat' => '37.6373008'
                           'importance' => 0.62
                           'addressimportance' => '9.297392099151168'
                           'extra_place' => 
                           'addresstype' => 'road'
                           'aBoundingBox' => 0 => '37.6360982'
                                             1 => '37.6383799'
                                             2 => '-122.1249293'
                                             3 => '-122.1249005'
                           'label' => 'Residential'
                           'name' => 'Viking Street, Hayward, Comté d'Alameda, Californie, 94545, États-Unis d'Amérique'
                           'foundorder' => -2.9276792099151
```

Where osm_type is W and on the official its T

**Reply by mtmail** [reply]
So the table `location_property_tiger` gets queried, that's good.

Can you check that the `location_property_tiger` has entries near the street and that the entries point to a `placex` street?
place_id and parent_place_id are different on every server, but the osm_type and osm_id should be the same.

```sql
SELECT count(*) FROM location_property_tiger WHERE postcode='94545';
1674

-- https://www.openstreetmap.org/way/6328881#map=19/37.63724/-122.12486
SELECT place_id, parent_place_id, startnumber, endnumber FROM location_property_tiger
WHERE postcode='94545' AND ST_DISTANCE(linegeo, 'SRID=4326;POINT(-122.12486 37.63724)'::geometry) < 0.0002;
 place_id  | parent_place_id | startnumber | endnumber
-----------+-----------------+-------------+-----------
 308837842 |       102359495 |       25000 |     25098
 308837839 |       102359495 |       25001 |     25079

SELECT osm_type, osm_id, name FROM placex WHERE place_id=102359495;
 osm_type | osm_id  |          name
----------+---------+-------------------------
 W        | 6328881 | "name"=>"Viking Street"
```

**Reply by ghost** [reply]
Here are my results:

```
nominatim=# SELECT count(*) FROM location_property_tiger WHERE postcode='94545';
 count
-------
  1653
(1 row)

nominatim=# SELECT place_id, parent_place_id, startnumber, endnumber FROM location_property_tiger
WHERE postcode='94545' AND ST_DISTANCE(linegeo, 'SRID=4326;POINT(-122.12486 37.63724)'::geometry) < 0.0002;
 place_id  | parent_place_id | startnumber | endnumber
-----------+-----------------+-------------+-----------
 346079325 |       102394391 |       25000 |     25098
 346079326 |       102394391 |       25001 |     25099
(2 rows)

nominatim=# SELECT osm_type, osm_id, name FROM placex WHERE place_id=102394391;
 osm_type | osm_id  |          name
----------+---------+-------------------------
 W        | 6328881 | "name"=>"Viking Street"
(1 row)
```

**Reply by ghost** [reply]
So it seems to be exactly the same, apart from the `endnumber` which I'm not sure what it represents.

**Reply by lonvia** [reply]
Tiger data lookup is not enabled. Try running `nominatim refresh --website`.

**Reply by ghost** [reply]
Fixed it by moving the .env file outside the import folder (my bad) and doing `nominatim refresh --website`, thank you both!

**Comment by brad-4k** [comment]
Can I jump in on this discussion?

I'm running into a similar issue but... worse.  I performed a whole world import as well and but there are some addresses where I get nothing in return (on my setup) but there is a decent answer from the "public" nominatim server.

It's not all addresses but a few.  How might I verify that my import is correct?

Query on my server:
http://localhost:8080/search?q=38+Abel+Tasmanstraat,Rotterdam,NL+3165&format=geojson

Answer:
```
{
  "type": "FeatureCollection",
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "features": [
    
  ]
}
```

Query on the public server:
https://nominatim.openstreetmap.org/search?q=38+Abel+Tasmanstraat,Rotterdam,NL+3165&format=geojson

Answer:
```
{
  "type": "FeatureCollection",
  "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
  "features": [
    {
      "type": "Feature",
      "properties": {
        "place_id": 101239458,
        "osm_type": "way",
        "osm_id": 7518933,
        "display_name": "Abel Tasmanstraat, Distripark Eemhaven, Rotterdam-Albrandswaard, Albrandswaard, South Holland, Netherlands, 3165 AM, Netherlands",
        "place_rank": 26,
        "category": "highway",
        "type": "unclassified",
        "importance": 0.5
      },
      "bbox": [
        4.4134436,
        51.8678575,
        4.4243001,
        51.8703737
      ],
      "geometry": {
        "type": "Point",
        "coordinates": [
          4.4161459,
          51.8688234
        ]
      }
    }
  ]
}
```

**Reply by mtmail** [reply]
Can you create a new discussion? There’s a couple of database queries we can try. Could also be an algorithm difference, nominatim.openstreetmap.org uses the master branch which is about 3 months ahead.

**Reply by brad-4k** [reply]
Will do...
